src/experiments.ts

```diff
@@ -30,41 +34,48 @@ export class ExperimentalSettings extends Disposable {
         private readonly logger: ILogService
     ) {
         super();
-
-        const configCatOptions = {
-            logger: {
-                debug(): void { },
-                log(): void { },
-                info(): void { },
-                warn(message: string): void { logger.warn(`ConfigCat: ${message}`); },
-                error(message: string): void { logger.error(`ConfigCat: ${message}`); }
-            },
-            requestTimeoutMs: 1500,
-            cacheTimeToLiveSeconds: 60
+        const clientLogger: configcatcommon.IConfigCatLogger = {
+            debug(): void { },
+            log(): void { },
+            info(): void { },
+            warn(message: string): void { logger.warn(`ConfigCat: ${message}`); },
+            error(message: string): void { logger.error(`ConfigCat: ${message}`); }
         };
-
         const gitpodHost = context.extensionMode === vscode.ExtensionMode.Production ? this.hostService.gitpodHost : 'https://gitpod-staging.com';
-        this.configcatClient = configcat.createClientWithLazyLoad(key, {
-            baseUrl: new URL('/configcat', gitpodHost).href,
-            ...configCatOptions
-        });
+        this.configcatClient = this.createClient(gitpodHost, key, clientLogger);
 
         this._register(hostService.onDidChangeHost(() => {
             this.configcatClient.dispose();
             const gitpodHost = context.extensionMode === vscode.ExtensionMode.Production ? this.hostService.gitpodHost : 'https://gitpod-staging.com';
-            this.configcatClient = configcat.createClientWithLazyLoad(key, {
-                baseUrl: new URL('/configcat', gitpodHost).href,
-                ...configCatOptions
-            });
+            this.configcatClient = this.createClient(gitpodHost, key, clientLogger);
         }));
-
         this.extensionVersion = new semver.SemVer(extensionVersion);
     }
 
+    createClient(gitpodHost: string, key: string, clientLogger: configcatcommon.IConfigCatLogger) {
+        let ignoredFirstChange = false;
+        return configcat.getClient(key, configcat.PollingMode.AutoPoll, {
```

Comment:
We need to change polling mode to `auto` otherwise hook on `configChanged` makes no sense.

Reply:
why do we need `configChanged` event?